src/service/resources-impl.js

```diff
@@ -1327,6 +1327,13 @@ export class Resources {
     // Breadth-first search.
     if (element.classList.contains('-amp-element')) {
       callback(this.getResourceForElement(element));
+      // Also schedule amp-element that is a placeholder for the element.
+      if (!element.classList.contains('-amp-layout')) {
+        const placeholder = element.getPlaceholder();
+        if (placeholder && placeholder.classList.contains('-amp-element')) {
```

Comment:
I misunderstood your comment initially but got it after you explained the required test part. Done.